src/downloader.js

```diff
@@ -270,4 +268,14 @@ class ImmediateDownload {
   }
 }
 
+
+class DownloadError extends Error {
+  constructor(failedDownloads) {
+    super('Download(s) failed; see `.failedDownloads`.');
+    this.name = this.constructor.name;
+    this.failedDownloads = failedDownloads;
+  }
+}
+window.DownloadError = DownloadError;
```

Comment:
If you want it to be public, just put it outside the scope that hides the private members.